examples/basic_cuda_example.py

```diff
@@ -0,0 +1,179 @@
+install them with conda. For example, to install FFmpeg version 7:
+
+.. code-block:: bash
+   # These libraries are needed for CUDA decoding with TorchCodec
+   conda install ffmpeg=7 -c conda-forge
+   conda install libnpp cuda-nvrtc -c nvidia
+"""
+
+# %%
+######################################################################
+# Checking if Pytorch has CUDA enabled
+######################################################################
+#
+# .. note::
+#
+#    This tutorial requires FFmpeg libraries compiled with CUDA support.
+#
+#
+import torch
+
+print(f"{torch.__version__=}")
+print(f"{torch.cuda.is_available()=}")
+print(f"{torch.cuda.get_device_properties(0)=}")
+
+
+# %%
+######################################################################
+# Downloading the video
+######################################################################
+#
+# We will use the following video which has the following properties;
+#
+# - Codec: H.264
+# - Resolution: 960x540
+# - FPS: 29.97
+# - Pixel format: YUV420P
+#
+# .. raw:: html
+#
+#    <video style="max-width: 100%" controls>
+#      <source src="https://download.pytorch.org/torchaudio/tutorial-assets/stream-api/NASAs_Most_Scientifically_Complex_Space_Observatory_Requires_Precision-MP4_small.mp4" type="video/mp4">
+#    </video>
+import urllib.request
+
+video_file = "video.mp4"
+urllib.request.urlretrieve(
+    "https://download.pytorch.org/torchaudio/tutorial-assets/stream-api/NASAs_Most_Scientifically_Complex_Space_Observatory_Requires_Precision-MP4_small.mp4",
+    video_file,
+)
+
+
+# %%
+######################################################################
+# CUDA Decoding using VideoDecoder
+######################################################################
+#
+# To use CUDA decoder, you need to pass in a cuda device to the decoder.
+#
+from torchcodec.decoders import VideoDecoder
+
+vd = VideoDecoder(video_file, device="cuda:0")
+frame = vd[0]
+
+# %%
+#
+# The video frames are decoded and returned as tensor of NCHW format.
+
+print(frame.data.shape, frame.data.dtype)
+
+# %%
+#
+# The video frames are left on the GPU memory.
+
+print(frame.data.device)
+
+
+# %%
+######################################################################
+# Visualizing Frames
+######################################################################
+#
+# Let's look at the frames decoded by CUDA decoder and compare them
+# against equivalent results from the CPU decoders.
+import matplotlib.pyplot as plt
+
+
+def get_frames(timestamps: list[float], device: str):
+    decoder = VideoDecoder(video_file, device=device)
+    return [decoder.get_frame_played_at(ts) for ts in timestamps]
+
+
+def get_numpy_images(frames):
+    numpy_images = []
+    for frame in frames:
+        # We transfer to the CPU so they can be visualized by matplotlib.
+        numpy_image = frame.data.to("cpu").permute(1, 2, 0).numpy()
+        numpy_images.append(numpy_image)
+    return numpy_images
```

Comment:
Consider using the same logic we have in our 2 existing examples for plotting images (and for plot_cpu_and_cuda_images).

https://github.com/pytorch/torchcodec/blob/c2bea4be0755aec1d9ca0813b2125c6f90f0bd8a/examples/basic_example.py#L36-L51

It'll be easier to clean up / factorize into a util later. You typically don't need to directly convert to numpy for this.

It will also make the example easier to follow because right now there are plotting/conversion utilities mixed with decoding code. It's best to keep those separate.

Reply:
I actually tried this suggestion but the results look very ugly. Something (sphinx, torchvision or matplotlib) is compressing the images and it makes them look really low-resolution.

Example code that I used and output:

```
def plot(
    frames1: List[torch.Tensor],
    frames2: List[torch.Tensor],
    title1: Optional[str] = None,
    title2: Optional[str] = None,
):
    try:
        import matplotlib.pyplot as plt
        from torchvision.transforms.v2.functional import to_pil_image
        from torchvision.utils import make_grid
    except ImportError:
        print("Cannot plot, please run `pip install torchvision matplotlib`")
        return
    # Create a list of alternating frames from frames1 and frames2
    frames = [f for pair in zip(frames1, frames2) for f in pair]
    plt.rcParams["savefig.bbox"] = "tight"
    fig, ax = plt.subplots()
    ax.imshow(to_pil_image(make_grid(frames, nrow=2)))
    ax.set(xticklabels=[], yticklabels=[], xticks=[], yticks=[])
    plt.tight_layout()
```

Output:

![image](https://github.com/user-attachments/assets/c3cedb12-71ae-4447-b6aa-ceec7af45fb9)


Original output:

![image](https://github.com/user-attachments/assets/06b929f6-4cd4-4fa6-9d16-2aea1881a7cf)


The original output is better so IMO we should submit as-is and iterate on this later (without compromising the image quality).